mediawiki/extensions/CASAuth/CASAuth.php

```diff
@@ -83,18 +83,20 @@ function casLogin($user, &$result) {
 						
 						$website_cookie = ".openfoodfacts.org";
 						$name_cookie = "session";
-						$cookie = $_COOKIE; // tu peux r�cup�rer le cookie en PHP
+						$cookie = $_COOKIE; // tu peux récupérer le cookie en PHP
 
 						/* 		user_session
 							&	[session_id]
 							& 	user_id
 							&	stephane 			*/
 						$valeur_cookie = $cookie[$name_cookie];
 						
-						//Parser user_session&4414725571&user_id&manu1400
-						$tab_valeurs = split("&", $valeur_cookie); //Pas de limite
-						$numero_session = $tab_valeurs[1]; //num�ro de la session
-						$username = $tab_valeurs[3]; //nom de l'utilisateur
+						//Parser user_session&4414725571&user_id&manu1400 using http://stackoverflow.com/a/32768029/11963
+						$chunks = array_chunk(preg_split('/&/', $valeur_cookie), 2);
+						$tab_valeurs = array_combine(array_column($chunks, 0), array_column($chunks, 1));
```

Comment:
Note that `array_column` requires PHP 5.5+. (PHP 5.6 is EOL for about half a year right now.)